3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,320 @@
+		}
+	}
+
+	/**
+	 * Extract image dimensions via wpcom/imagesize.
+	 *
+	 * @since 0.5
+	 *
+	 * @param array $dimensions Dimensions.
+	 * @return array Dimensions.
+	 */
+	static function extract_image_dimensions_from_getimagesize( $dimensions ) {
+		if ( ! ( defined('WPCOM') && WPCOM && function_exists( 'require_lib' ) ) ) {
+			return $dimensions;
+		}
+		require_lib( 'wpcom/imagesize' );
+
+		foreach ( $dimensions as $url => $value ) {
+			if ( is_array( $value ) ) {
+				continue;
+			}
+			$result = wpcom_getimagesize( $url );
+			if ( is_array( $result ) ) {
+				$dimensions[ $url ] = array(
+					'width'  => $result[0],
+					'height' => $result[1],
+				);
+			}
+		}
+
+		return $dimensions;
+	}
+
+	static function amp_post_jetpack_og_tags() {
+		Jetpack::init()->check_open_graph();
+		if ( function_exists( 'jetpack_og_tags' ) ) {
+			jetpack_og_tags();
+		}
+	}
+
+	static function videopress_enable_freedom_mode( $options ) {
+		$options['freedom'] = true;
+		return $options;
+	}
+
+	static function render_sharing_html( $markup, $sharing_enabled ) {
+		remove_action( 'wp_footer', 'sharing_add_footer' );
+		if ( empty( $sharing_enabled ) ) {
+			return $markup;
+		}
+		$supported_services = array(
+			'facebook'      => array(
+				/** This filter is documented in modules/sharedaddy/sharing-sources.php */
+				'data-param-app_id' => apply_filters( 'jetpack_sharing_facebook_app_id', '249643311490' ),
+			),
+			'twitter'       => array(),
+			'pinterest'     => array(),
+			'whatsapp'      => array(),
+			'google-plus-1' => array(
+				'type' => 'gplus',
+			),
+			'tumblr'        => array(),
+			'linkedin'      => array(),
+		);
+		$sharing_links = array();
+		foreach ( $sharing_enabled['visible'] as $id => $service ) {
+			if ( ! isset( $supported_services[ $id ] ) ) {
+				$sharing_links[] = "<!-- not supported: $id -->";
+				continue;
+			}
+			$args = array_merge(
+				array(
+					'type' => $id,
+				),
+				$supported_services[ $id ]
+			);
+			$sharing_link = '<amp-social-share';
+			foreach ( $args as $key => $value ) {
+				$sharing_link .= sprintf( ' %s="%s"', sanitize_key( $key ), esc_attr( $value ) );
+			}
+			$sharing_link .= '></amp-social-share>';
+			$sharing_links[] = $sharing_link;
+		}
+		return preg_replace( '#(?<=<div class="sd-content">).+?(?=</div>)#s', implode( '', $sharing_links ), $markup );
+	}
+}
+
+add_action( 'init', array( 'Jetpack_AMP_Support', 'init' ), 1 );
+
+add_action( 'admin_init', array( 'Jetpack_AMP_Support', 'admin_init' ), 1 );
+
+// this is necessary since for better or worse Jetpack modules and widget files are loaded during plugins_loaded, which means we must
+// take the opportunity to intercept initialisation before that point, either by adding explicit detection into the module,
+// or preventing it from loading in the first place (better for performance)
+add_action( 'plugins_loaded', array( 'Jetpack_AMP_Support', 'init_filter_jetpack_widgets' ), 1 );
```

Comment:
Minor nitpick: There should an empty newline at the end of the file.

Reply:
Fixed in new PR